src/Ads/AssetSuggestionsService.php

```diff
@@ -17,13 +19,181 @@
+		}
+
+		$gallery_images_urls = get_post_gallery_images( $id );
+		$marketing_images    = array_merge( $this->get_url_attachments_by_ids( $attachments_ids ), $gallery_images_urls );
+		$marketing_images    = array_slice( $marketing_images, 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES );
+		$long_headline       = get_bloginfo( 'name' ) . ': ' . $post->post_title;
+
+		return [
+			'headline'                => [ $post->post_title ],
+			'long_headline'           => [ $long_headline ],
+			'description'             => $this->remove_empty_values( [ $post->post_excerpt, get_bloginfo( 'description' ) ] ),
+			'logo'                    => $this->remove_empty_values( [ wp_get_attachment_image_url( get_theme_mod( 'custom_logo' ) ) ] ),
+			'final_url'               => get_permalink( $id ),
+			'business_name'           => get_bloginfo( 'name' ),
+			'display_url_path'        => [ $post->post_name ],
+			'square_marketing_images' => $marketing_images,
+			'marketing_images'        => $marketing_images,
+			'call_to_action'          => null,
+		];
+	}
+
+	/**
+	 * Get attachments related to the shop page.
+	 *
+	 * @return array Shop attachments.
+	 */
+	protected function get_shop_attachments(): array {
+		return $this->get_post_attachments(
+			[
+				'post_parent__in' => $this->get_shop_products(),
+			]
+		);
+	}
+
+	/**
+	 *
+	 * Get products that will be use to offer image assets.
+	 *
+	 * @param array $args See WP_Query::parse_query() for all available arguments.
+	 * @return array Shop products.
+	 */
+	protected function get_shop_products( array $args = [] ): array {
+		$defaults = [
+			'post_type'   => 'product',
+			'numberposts' => self::DEFAULT_MAXIMUM_MARKETING_IMAGES,
+			'fields'      => 'ids',
+		];
+
+		$args = wp_parse_args( $args, $defaults );
+
+		return $this->wp->get_posts( $args );
+	}
+
+	/**
+	 * Remove empty values from array.
+	 *
+	 * @param array $array A list of strings.
+	 *
+	 * @return array A list of strings without empty strings.
+	 */
+	protected function remove_empty_values( array $array ): array {
+		return array_values( array_filter( $array ) );
+	}
+
+	/**
+	 * Get URL for each attachment using an array of attachment ids.
+	 *
+	 * @param array $ids A list of attachments ids.
+	 *
+	 * @return array A list of attachments urls.
+	 */
+	protected function get_url_attachments_by_ids( array $ids ): array {
+		$ids = array_unique( $this->remove_empty_values( $ids ) );
+
+		$marketing_images = [];
+		foreach ( $ids as $id ) {
+			$marketing_images[] = wp_get_attachment_image_url( $id );
+		}
+		return $marketing_images;
+	}
+
+
+	/**
+	 * Get Attachmets for specific posts.
+	 *
+	 * @param array $args See WP_Query::parse_query() for all available arguments.
+	 *
+	 * @return array List of attachments
+	 */
+	protected function get_post_attachments( array $args = [] ): array {
+		$defaults = [
+			'post_type'      => 'attachment',
+			'post_mime_type' => 'image',
```

Comment:
💅 We are only getting `image` mime type in this method, maybe rename the method name to `get_post_image_attachments`?